### PR TITLE
Add push switch to builders

### PR DESCRIPTION
**Pull request by r2d4**
Add an option to skip pushing the docker image for testing or local clusters like minikube or docker for desktop.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/67)

### CODE REVIEW

File: fairing/builders/cluster/gcs_context.py [hunk 1]
```diff
@@ -54,7 +54,12 @@ def cleanup(self):
             self.created_secret.metadata.namespace,
             client.V1DeleteOptions())
 
-    def generate_pod_spec(self, image_name):
+    def generate_pod_spec(self, image_name, push):
+        args = ["--dockerfile=Dockerfile",
+                          "--destination=" + image_name,
+                          "--context=" + self.uploaded_context_url]
+        if not push:
+            args.append("--no-push")
```
Comment (karthikv2k) [inline fairing/builders/cluster/gcs_context.py]: For no push case for cluster builder, where does the built image live? I am not able understand a valid use case for this.

Comment (r2d4) [reply fairing/builders/cluster/gcs_context.py]: Testing/parity with other builders. It’s mainly functional for the docker daemon alone right now, but the append builder and cluster should optionally perform a docker load to have the same effect

**Pull request merged**